src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1812,6 +1812,10 @@ private function addRateLimiterSection(ArrayNodeDefinition $rootNode)
                     ->info('Rate limiter configuration')
                     ->{!class_exists(FullStack::class) && class_exists(TokenBucketLimiter::class) ? 'canBeDisabled' : 'canBeEnabled'}()
                     ->fixXmlConfig('limiter')
+                    ->beforeNormalization()
+                        ->ifTrue(function ($v) { return \is_array($v) && !isset($v['enabled']); })
+                        ->then(function ($v) { return $v + ['enabled' => true]; })
```

Comment:
This is not the right fix: the issue is the next `beforeNormalisation` block (5 lines below)
```diff
 $newV = [
-    'enabled' => $v['enabled'],
+    'enabled' => $v['enabled'] ?? true,
 ];

Reply:
I just followed the same config that the Lock section uses


https://github.com/symfony/symfony/blob/5.2/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1104-L1116

Reply:
mhh nop, `true` should not be the default value, as it can be enable or disabled.
```php
if (isset($v['enabled'])) {
    $newV = [
        'enabled' => $v['enabled'],
    ];
    unset($v['enabled']);
} else {
    $newV = [];
}
```

Reply:
`true` will only be the default value when not using fullstack.

In that case, the config uses `canBeDisabled`, which implies it is enabled by default (and which is also the case where the `enabled` key is not set).

When not using fullstack, the config uses `canBeEnabled`, which then sets the `enabled` key to `false` b y default

Reply:
@pierredup but this runs only if the config is an array, and so it has been configured explicitly. In such case, we want it to be enabled.

Reply:
Also, the `enabled` key does need to be set on the `$newV` variable, otherwise the call at https://github.com/symfony/symfony/blob/5.2/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L424 will fail. So if `enabled` is not set, then it does need a default value

Reply:
> but this runs only if the config is an array, and so it has been configured explicitly. In such case, we want it to be enabled.

@stof This is exactly what this PR fixes. If you configure the rate limiter, but don't set the `enabled: true` key, then it needs to be enabled by default. Or maybe I'm not understanding your comment

Reply:
The correct fix here is to change line 1819 (1823 in this PR) to:

```diff
-     'enabled' => $v['enabled'],
+     'enabled' => $v['enabled'] ?? !class_exists(FullStack::class) && class_exists(TokenBucketLimiter::class),
```

and then remove this new beforeNormalization function.

I've just tested this in a small reproducer and it seems to result in the correct settings. We should never hard code `true`, not in this function nor in the function below, as it depends on what is installed.

Reply:
Sorry for the noise, @stof and @jderusse are correct here. 

And to be honest, I think we all three agree! However, @pierredup is misunderstanding the fix. Instead of adding a new `beforeNormalization` block to fix this case, @jderusse proposed to directly fix this on line 1819:
```diff
  $newV = [
-     'enabled' => $v['enabled'],
+     'enabled' => $v['enabled'] ?? true,
  ]
```

This fixes all cases that you are describing, and removes the need for yet another normalizer.

Reply:
Thanks @wouterj, I don't misunderstand the fix, and I didn't disagree with that fix, I just pointed out that I used the same convention that the lock component uses to set the default value of `enabled`. That is why I used the beforeNormalize method, to be consistent with other places in the code